qabelbox/src/main/java/de/qabel/qabelbox/chat/interactor/TransformingChatUseCase.kt

```diff
@@ -31,13 +33,37 @@ class TransformingChatUseCase @Inject constructor(val identity: Identity, overri
     }
 
     override fun retrieve() = observable<ChatMessage> { subscriber ->
-        thread {
+        chatDropMessageRepository.markAsRead(contact, identity)
+        chatDropMessageRepository.findByContact(contact.id, identity.id).forEach { msg ->
+            subscriber.onNext(chatMessageTransformer.transform(msg))
+        }
+        subscriber.onCompleted()
+    }
```

Comment:
I think it is easier to create the observable directly from the findByContact method and subscribe to your own onComplete to mark them as read.